src/app/customer/CreateAccountForm.tsx

```diff
@@ -0,0 +1,119 @@
+import { FormField } from '@bigcommerce/checkout-sdk';
+import { withFormik, FormikProps } from 'formik';
+import { noop } from 'lodash';
+import React, { FunctionComponent } from 'react';
+
+import { preventDefault } from '../common/dom';
+import { withLanguage, TranslatedString, WithLanguageProps } from '../locale';
+import { Alert, AlertType } from '../ui/alert';
+import { Button, ButtonVariant } from '../ui/button';
+import { DynamicFormField, Fieldset, Form } from '../ui/form';
+
+import getCreateCustomerValidationSchema, { CreateAccountFormValues } from './getCreateCustomerValidationSchema';
+import getPasswordRequirements from './getPasswordRequirements';
+import './CreateAccountForm.scss';
+
+export interface CreateAccountFormProps {
+    formFields: FormField[];
+    createAccountError?: Error;
+    isCreatingAccount?: boolean;
+    createAccountLink: string;
+    shouldRender: boolean;
+    onCancel?(): void;
+    onSubmit?(values: CreateAccountFormValues): void;
+}
+
+const CreateAccountForm: FunctionComponent<CreateAccountFormProps & WithLanguageProps & FormikProps<CreateAccountFormValues>> = ({
+    formFields,
+    createAccountError,
+    isCreatingAccount,
+    onCancel,
+    shouldRender,
+    createAccountLink,
+}) => {
+
+    if (!shouldRender) {
+        window.top.location.assign(createAccountLink);
```

Comment:
We should avoid performing any side effect in a `render` function. Doing a redirect is considered to be a side effect as it causes a direct change external to the function itself. You can use `useEffect` to perform a side effect in a `render` function but I wouldn't recommend in this case because you'll end up writing something like "do a redirect every time `shouldRender` is changed to `true` or the component is mounted". A better place to perform a redirect is inside an event handler or whatever callback that's triggered when the user clicks on "create account" link (e.g.: `onCreateAccount`). In there, you can determine whether to show a form or do a render based on the config value.

Reply:
Sorry I thought this was different to doing a redirect inside a render function because this is an FC, but I get your point, will readdress.

Reply:
@davidchin pushed some changes earlier,  hopefully I got it right this time